core/api/src/main/java/org/trellisldp/api/BinaryService.java

```diff
@@ -36,34 +36,35 @@
      * @param identifier an identifier used for locating the binary object
      * @param from the starting point of a range request
      * @param to the ending point of a range request
-     * @return the content
+     * @return the new completion stage with the binary content
      */
-    Optional<InputStream> getContent(IRI identifier, Integer from, Integer to);
+    CompletableFuture<InputStream> getContent(IRI identifier, Integer from, Integer to);
 
     /**
      * Get the content of the binary object.
      *
      * @param identifier an identifier used for locating the binary object
-     * @return the content
+     * @return the new completion stage with the binary content
      */
-    Optional<InputStream> getContent(IRI identifier);
+    CompletableFuture<InputStream> getContent(IRI identifier);
 
     /**
      * Test whether a binary object exists at the given URI.
      *
      * @param identifier the binary object identifier
-     * @return whether the binary object exists
+     * @return the new completion stage, indicating whether the binary object exists
      */
-    Boolean exists(IRI identifier);
+    CompletableFuture<Boolean> exists(IRI identifier);
```

Comment:
This method can be removed entirely. It is not used anywhere in the codebase, and keeping it here just puts extra burden on implementations of this interface.